src/DataBuilder.php

```diff
@@ -1193,19 +1213,15 @@ public function detectGitBranch($allowExec = true)
         return null;
     }
     
-    private static function getGitBranch()
-    {
-        try {
-            if (function_exists('shell_exec')) {
-                $stdRedirCmd = Utilities::isWindows() ? " > NUL" : " 2> /dev/null";
-                $output = rtrim(shell_exec('git rev-parse --abbrev-ref HEAD' . $stdRedirCmd));
-                if ($output) {
-                    return $output;
-                }
+    private static function getGitBranch(): ?string
+    {
+        if (function_exists('shell_exec')) {
+            $stdRedirCmd = Utilities::isWindows() ? ' > NUL' : ' 2> /dev/null';
+            $output = shell_exec('git rev-parse --abbrev-ref HEAD' . $stdRedirCmd);
+            if (is_string($output)) {
+                return rtrim($output);
             }
-            return null;
-        } catch (\Exception $e) {
-            return null;
         }
+        return null;
```

Comment:
I think this can be simplified after refactoring a null logic:

```php
    private static function getGitBranch(): ?string
    {
        if (!function_exists('shell_exec')) return null;

        $stdRedirCmd = Utilities::isWindows() ? ' > NUL' : ' 2> /dev/null';
        $output = shell_exec('git rev-parse --abbrev-ref HEAD' . $stdRedirCmd);
        if (!$output) return null;
        return rtrim($output);
    }
```

Regarding removing try/catch block - we don't need to clean this up in case of occurring `E_WARNING` for `shell_exec`, correct?

Reply:
I didn't want to stray too far from the style of the original code. The changes here are meant to be obviously faithful to the original so as to not obscure the intent of my change (namely fixing psalm issues, not style). That said, the suggestion doesn't pass the linting rules:
```
FILE: /opt/rollbar/rollbar-php/src/DataBuilder.php
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 1218 | ERROR | [x] Inline control structures are not allowed
 1222 | ERROR | [x] Inline control structures are not allowed
```

> Regarding removing try/catch block - we don't need to clean this up in case of occurring E_WARNING for shell_exec, correct?

Not in this changeset, no. There are plenty of places where the code ignores the raised errors, polutting the users logs. Those need a pass.

Reply:
It's not about the style of the code. It's all about the wrong design of the logic. If the original code is of bad quality, I don't think we should reuse the wrong patterns.

Your implementation does not provide a clean solution for possibly null arguments. The code doesn't self-document, provide rationales for the next steps, etc. It is difficult to read/analyze and may reduce maintainability in the future. The code I suggest follows the design pattern of composing method.

> That said, the suggestion doesn't pass the linting rules:

I don't know the lining rules but the output clearly shows how to fix it:

```php
    private static function getGitBranch(): ?string
    {
        if (!function_exists('shell_exec')) {
            return null;
        }

        $stdRedirCmd = Utilities::isWindows() ? ' > NUL' : ' 2> /dev/null';
        $output = shell_exec('git rev-parse --abbrev-ref HEAD' . $stdRedirCmd);
        if (!$output) {
            return null;
        }
        return rtrim($output);
    }
```